packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js

```diff
@@ -214,12 +214,6 @@ if (__DEV__) {
       return true;
     }
 
-    // Now that we've validated casing, do not validate
-    // data types for reserved props
-    if (isReserved) {
```

Comment:
I decided to remove this since the `shouldRemoveAttributeWithWarning` function already duplicates this logic internally.